src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -2489,14 +2493,45 @@ void GraphOptimizer::MergeTransposeAndReorder(Graph &graph) {
         auto reorderNode = graph.InsertReorder(edge, reorderlayerName, *reorderInDesc, *reorderOutDesc, isOptimized, srcPerm);
 
         // case 2
+        auto finalReorder = reorderNode;
         if (inPrec != outPrec) {
             auto reorderInDesc2 = reorderOutDesc;
             auto reorderOutDesc2 = outDesc;
 
             std::string reorderLayerName2 = reorderNode->getName() + "_" +
                                     Reorder::getReorderArgs(*reorderInDesc2, *reorderOutDesc2) + "_" + childChildNode->getName();
 
-            graph.InsertReorder(reorderNode->getChildEdgeAt(0), reorderLayerName2, *reorderInDesc2, *reorderOutDesc2, false);
+            finalReorder = graph.InsertReorder(reorderNode->getChildEdgeAt(0), reorderLayerName2, *reorderInDesc2, *reorderOutDesc2, false);
```

Comment:
Why not just extend the insertReoder to insert reorders to all the children ? Inserting one reorder then connecting to others needs more checks.

Reply:
good idea, but `insertReoder`'s API limited this possibility, I will try to avoid using `insertReoder` and refactor this logic according to your idea!

Reply:
this block of code is refactored according to your comment, with the help of slightly adjusted `Graph::InsertNode` which allows initializing node w/o creating parent & child edges, .